File: src/main/java/com/busted_moments/client/screen/territories/ManageTerritoriesScreen.java
package com.busted_moments.client.screen.territories;

import com.busted_moments.client.models.territory.eco.TerritoryEco;
import com.busted_moments.client.models.territory.eco.TerritoryScanner;
import com.busted_moments.client.models.war.WarModel;
import com.busted_moments.client.util.ContainerHelper;
import com.busted_moments.client.util.SoundUtil;
import com.wynntils.utils.mc.McUtils;
import net.minecraft.sounds.SoundEvents;
import net.minecraft.sounds.SoundSource;
import net.minecraft.world.item.ItemStack;

import java.util.regex.Pattern;

import static com.busted_moments.client.util.Textures.TerritoryMenu.BACKGROUND;

public class ManageTerritoriesScreen extends TerritoryScreen<ManageTerritoriesScreen.Scanner> {
   public static final Pattern TERRITORY_MENU_PATTERN = Pattern.compile("^(?<guild>.+): Territories$");
   public static final int LOADOUTS_SLOT = 36;

   public ManageTerritoriesScreen(int id, boolean showProduction, boolean showPercents) {
      super(id, showProduction, true, showPercents);
   }

   @Override
   protected Scanner scanner(int container) {
      return new Scanner(container);
   }

   @Override
   protected Entry entry(TerritoryEco territory) {
      return new Entry(territory);
   }

   @Override
   protected Pattern title() {
      return TERRITORY_MENU_PATTERN;
   }

   @Override
   protected void build() {
      item(LOADOUTS_SLOT, false, true)
              .setScale(1.05F)
              .perform(item -> {
                 item.setX(2.75F + (this.width / 2F) - BACKGROUND.width() / 2F);
                 item.setY(((this.height / 2F) + BACKGROUND.height() / 2F) - item.getHeight() - 3);
              }).build();
   }

   public class Entry extends AbstractEntry {
      public Entry(TerritoryEco territory) {
         super(territory);
      }

      @Override
      protected void click() {
         BUSY = true;
         scanner.select(getItemSupplier().get());
         SoundUtil.play(SoundEvents.WOODEN_PRESSURE_PLATE_CLICK_ON, SoundSource.MASTER, 1, 1);
      }
   }

   public static class Scanner extends TerritoryScanner {
      private String TO_SELECT = null;

      public Scanner(int containerId) {
         super(containerId);
      }

      public void select(ItemStack stack) {
         TO_SELECT = TerritoryEco.getTerritory(stack);

         if (WarModel.current().isEmpty()) {
            close();
            McUtils.sendCommand("gu territory %s".formatted(TO_SELECT));
         }

         if (getPages().size() == 1) rescan();
      }

      @Override
      protected boolean process(String territory, ItemStack stack, int slot) {
         if (TO_SELECT != null && TO_SELECT.equals(territory)) {
            ContainerHelper.Click(slot, 0, containerId);
            return true;
         }

         return false;
      }
   }
}

File: src/main/kotlin/com/busted_moments/client/screens/territories/ManageTerritoriesScreen.kt
package com.busted_moments.client.screens.territories

import com.busted_moments.client.framework.Commands
import com.busted_moments.client.framework.render.elements.TextureElement
import com.busted_moments.client.framework.render.helpers.Percentage.Companion.pct
import com.busted_moments.client.framework.render.screen.Title
import com.busted_moments.client.framework.render.screen.elements.click
import com.busted_moments.client.framework.render.screen.elements.item
import com.busted_moments.client.framework.sounds.Sounds.play
import com.busted_moments.client.models.territories.eco.TerritoryData
import com.busted_moments.client.models.territories.eco.TerritoryScanner
import com.busted_moments.client.models.territories.war.WarModel
import net.essentuan.esl.time.duration.Duration
import net.essentuan.esl.time.duration.ms
import net.minecraft.sounds.SoundEvents
import net.minecraft.sounds.SoundSource
import net.minecraft.world.item.ItemStack

private const val LOADOUT_SLOT = 36

@Title("Manage Territories")
class ManageTerritoriesScreen(
    container: Int
) : TerritoryScreen<ManageTerritoriesScreen.Scanner>(
    container,
    Scanner(container)
) {
    init {
        scanner.screen = this
    }

    override fun select(territory: TerritoryData) {
        if (WarModel.current == null) {
            SoundEvents.WOODEN_PRESSURE_PLATE_CLICK_ON.play(SoundSource.MASTER)
            Commands.execute("gu territory ${territory.name}")
        } else
            scanner.select(territory)
    }

    override fun TextureElement<Context>.renderSidebar(
        ctx: Context
    ) {
        item {
            x = this@renderSidebar.x + ((50.pct of this@renderSidebar.width) - (50.pct of this.width)) - 0.5f
            y = (50.pct of ctx.window.guiScaledHeight) - (50.pct of this.height)

            stack = scanner.contents.getOrNull(BACK_SLOT) ?: return@item false
            tooltip = true

            click { _, _, _ ->
                scanner.click(BACK_SLOT, sound = SoundEvents.WOODEN_PRESSURE_PLATE_CLICK_ON)

                true
            }

            true
        }

        item {
            x = this@renderSidebar.x + ((50.pct of this@renderSidebar.width) - (50.pct of this.width))
            y = this@renderSidebar.y + this@renderSidebar.height - 5 - this.height

            stack = scanner.contents.getOrNull(LOADOUT_SLOT) ?: return@item false
            tooltip = true

            click { _, _, _ ->
                scanner.click(LOADOUT_SLOT)

                true
            }

            true
        }
    }

    class Scanner(container: Int) : TerritoryScanner(container) {
        internal lateinit var screen: ManageTerritoriesScreen
        private var selected: String? = null

        fun select(territory: TerritoryData) {
            selected = territory.name
        }

        override val delay: Duration
            get() = 0.ms

        override suspend fun process(territory: String, stack: ItemStack, slot: Int): Boolean {
            if (territory == selected) {
                selected = null
                click(slot, sound = SoundEvents.WOODEN_PRESSURE_PLATE_CLICK_ON)

                return true
            }

            return false
        }
    }
}